### PR TITLE
Fixed issues #595, #600, #606, #607, #609, #610, #620, #621

### PR DESCRIPTION
**Pull request by imran-ansari**

### CODE REVIEW

**Comment by imran-ansari** [comment]
@wdurairaj , @nilangekarss - Please review the changes.

File: hpedockerplugin/hpe/hpe_3par_mediator.py [hunk 1]
```diff
@@ -1033,101 +588,14 @@ def create_vfs(self, vfs_name, ip, subnet, cpg=None, fpg=None,
         finally:
             self._wsapi_logout()
 
-    def set_ACL(self, fMode, fUserId, fUName, fGName):
```
Comment (nilangekarss) [inline hpedockerplugin/hpe/hpe_3par_mediator.py]: Where is this code moved?

File: hpedockerplugin/file_manager.py [hunk 2]
```diff
@@ -447,35 +558,39 @@ def remove_share(self, share_name, share):
...
+        share.pop("id")
+        share.pop("readonly")
+        share.pop("comment")
+        LOG.info("Implementation details removed: %s" % share)
+        return share
 
+    @staticmethod
+    def get_share_details(share_name, db_share):
+        # TODO: mount_dir to be fixed later
         path_info = db_share.get('share_path_info')
```
Comment (nilangekarss) [inline hpedockerplugin/file_manager.py]: Do we have 'share_path_info' key if not this needs to be corrected?

File: hpedockerplugin/hpe/hpe_3par_mediator.py (hunk 1 above)
Comment (imran-ansari) [reply hpedockerplugin/hpe/hpe_3par_mediator.py]: This was an inadvertent removal while removing unused code. I've put it back

File: hpedockerplugin/file_manager.py (hunk 2 above)
Comment (imran-ansari) [reply hpedockerplugin/file_manager.py]: We don't... This will be addressed as part of #609

Comment (imran-ansari) [reply hpedockerplugin/file_manager.py]: This has been addressed in this PR only.

File: hpedockerplugin/cmd/cmd_deleteshare.py [hunk 3]
```diff
@@ -21,20 +20,61 @@ def __init__(self, file_mgr, share_info):
...
-                remaining_cnt = self._decrement_share_cnt()
-                if remaining_cnt == 0:
-                    self._delete_fpg()
+                self._decrement_share_cnt()
+
+            # If shares are not present on FPG after this delete, then
+            # delete the FPG too.
+            # WARNING: THIS WILL DELETE LEGACY FPG TOO IF IT BECOMES EMPTY
+            if not self._mediator.shares_present_on_fpg(self._fpg_name):
+                self._delete_fpg()
```
Comment (nilangekarss) [inline hpedockerplugin/cmd/cmd_deleteshare.py]: Here we can check whether fpg is legacy or Docker created FPG by querying etcd data base. If FPG is legacy FPG we will delete only etcd entry if present, if it is docker fpg, only then we will delete etcd entry and real FPG from 3PAR. As discussed over the call we will take it as separate PR. Rest of changes looks fine for me. Hence will approve this pull request and will merge.

1. If share is on legacy FPG, FPG will not be deleted , this has to be done manually by Administrator.
2. IF docker created FPG has stores on it while deleting docker created last share FPG will not be deleted. It needs to be cleaned by Administrator.

**Comment by nilangekarss** [comment]
These changes looks fine for me. These changes will address various bugs. Merging this PR.

**Pull request merged**